src/kakarot/kakarot.cairo

```diff
@@ -15,18 +15,38 @@ from kakarot.memory import Memory
 @constructor
 func constructor{
     syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, bitwise_ptr: BitwiseBuiltin*
-}(owner: felt, native_token_address_: felt) {
-    return Kakarot.constructor(owner, native_token_address_);
+}(owner: felt, native_token_address_: felt, evm_contract_class_hash: felt) {
+    return Kakarot.constructor(owner, native_token_address_, evm_contract_class_hash);
 }
 
-@external
+@view
 func execute{
     syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, bitwise_ptr: BitwiseBuiltin*
 }(code_len: felt, code: felt*, calldata_len: felt, calldata: felt*) -> (
+    stack_len: felt, stack: Uint256*, memory_len: felt, memory: felt*, gas_used: felt
+) {
+    alloc_locals;
+    let context = Kakarot.execute(code_len=code_len, code=code, calldata=calldata);
+    let len = Stack.len(context.stack);
+    return (
+        stack_len=len,
+        stack=context.stack.elements,
+        memory_len=context.memory.bytes_len,
+        memory=context.memory.bytes,
+        gas_used=context.gas_used,
+    );
+}
+
+// Create new function
+@external
+func execute_at_address{
+    syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, bitwise_ptr: BitwiseBuiltin*
+}(address: felt, calldata_len: felt, calldata: felt*) -> (
     stack_len: felt, stack: Uint256*, memory_len: felt, memory: felt*
 ) {
     alloc_locals;
-    let context = Kakarot.execute(code=code, code_len=code_len, calldata=calldata);
+
+    let context = Kakarot.execute_at_address(address=address, calldata=calldata);
```

Comment:
should we stop "cheating" and passing `calldata_len` to `execute_at_address`  instead of using a hint to compute its length?

Reply:
not sure about the reasons that motivated this, however, not sure it's related to this merge PR as well, but worth having the discussion @abdelhamidbakhta @LucasLvy any why regarding this helper?

Reply:
`let context = Kakarot.execute_at_address(address=address, calldata_len=calldata_len, calldata=calldata);` would be cleaner 
But we can flag it as a TODO and move forward

Reply:
I'd rather have @abdelhamidbakhta and @LucasLvy point of view regarding this as before we make any change as they AFAIK made this decision before and we're just consistent with it